test/test-basic.js

```diff
@@ -296,6 +296,51 @@ test('requestTimeout opt', async t => {
   }
 })
 
+test('concurrent requests opt', async t => {
+  const bootstrap = await setupTestnet(t)
+  const { rpcClient } = getRpcClient(t, bootstrap, { maxConcurrent: 4 })
+  const { server } = await setupRpcServer(t, bootstrap, { msDelay: 1000 })
+
+  let requestFinishedCount = 0
+
+  new Array(6).fill(0).map(async () => {
+    const res = await rpcClient.makeRequest(server.publicKey, 'echo', 'hi', {
+      requestEncoding: cenc.string,
+      responseEncoding: cenc.string
+    })
+    t.is(res, 'hi', 'rpc request processed successfully')
+    requestFinishedCount++
+  })
+
+  await new Promise(resolve => setTimeout(resolve, 1100))
+
+  t.is(requestFinishedCount, 4, 'only 4 requests finished')
+
+  await new Promise(resolve => setTimeout(resolve, 1000))
+
+  t.is(requestFinishedCount, 6, 'all requests finished')
+})
+
+test('concurrent requests opt with timeout (timeout release request count)', async t => {
+  const bootstrap = await setupTestnet(t)
+  const { rpcClient } = getRpcClient(t, bootstrap, { maxConcurrent: 4, requestTimeout: 1_000 })
+  const { server } = await setupRpcServer(t, bootstrap, { msDelay: 2_000 })
+
+  let requestFinishedCount = 0
+
+  new Array(6).fill(0).map(async (_, i) => {
```

Comment:
(detail) I don't think the `i` arg is used anywhere, so can be removed
Interesting test pattern btw, it works well here.

Question: in terms of the underlying behaviour, what happens here is:
- 4 requests are sent, the 2 others are queued
- after 1s, the 6 timeouts trigger. The 2 queued requests are never sent
- after another 1s, the server finishes the 4 requests